packages/api/src/beacon/routes/beacon/pool.ts

```diff
@@ -74,125 +101,212 @@ export type Api = {
-   * @throws ApiError
    */
-  submitPoolAttestations(
-    attestations: phase0.Attestation[]
-  ): Promise<ApiClientResponse<{[HttpStatusCode.OK]: void}, HttpStatusCode.BAD_REQUEST>>;
+  submitPoolAttestations: Endpoint<
+    "POST",
+    {signedAttestations: AttestationList},
+    {body: unknown},
+    EmptyResponseData,
+    EmptyMeta
+  >;
 
   /**
    * Submit AttesterSlashing object to node's pool
    * Submits AttesterSlashing object to node's pool and if passes validation node MUST broadcast it to network.
-   * @param requestBody
-   * @returns any Success
-   * @throws ApiError
    */
-  submitPoolAttesterSlashings(
-    slashing: phase0.AttesterSlashing
-  ): Promise<ApiClientResponse<{[HttpStatusCode.OK]: void}, HttpStatusCode.BAD_REQUEST>>;
+  submitPoolAttesterSlashings: Endpoint<
+    "POST",
+    {attesterSlashing: phase0.AttesterSlashing},
+    {body: unknown},
+    EmptyResponseData,
+    EmptyMeta
+  >;
 
   /**
    * Submit ProposerSlashing object to node's pool
    * Submits ProposerSlashing object to node's pool and if passes validation  node MUST broadcast it to network.
-   * @param requestBody
-   * @returns any Success
-   * @throws ApiError
    */
-  submitPoolProposerSlashings(
-    slashing: phase0.ProposerSlashing
-  ): Promise<ApiClientResponse<{[HttpStatusCode.OK]: void}, HttpStatusCode.BAD_REQUEST>>;
+  submitPoolProposerSlashings: Endpoint<
+    "POST",
+    {proposerSlashing: phase0.ProposerSlashing},
+    {body: unknown},
+    EmptyResponseData,
+    EmptyMeta
+  >;
 
   /**
    * Submit SignedVoluntaryExit object to node's pool
    * Submits SignedVoluntaryExit object to node's pool and if passes validation node MUST broadcast it to network.
-   * @param requestBody
-   * @returns any Voluntary exit is stored in node and broadcasted to network
-   * @throws ApiError
    */
-  submitPoolVoluntaryExit(
-    exit: phase0.SignedVoluntaryExit
-  ): Promise<ApiClientResponse<{[HttpStatusCode.OK]: void}, HttpStatusCode.BAD_REQUEST>>;
+  submitPoolVoluntaryExit: Endpoint<
+    "POST",
+    {signedVoluntaryExit: phase0.SignedVoluntaryExit},
+    {body: unknown},
+    EmptyResponseData,
+    EmptyMeta
+  >;
 
   /**
-   * Submit SignedBLSToExecutionChange object to node's pool
-   * Submits SignedBLSToExecutionChange object to node's pool and if passes validation node MUST broadcast it to network.
-   * @param requestBody
-   * @returns any BLSToExecutionChange is stored in node and broadcasted to network
-   * @throws ApiError
+   * Submit SignedBLSToExecutionChange objects to node's pool
+   * Submits SignedBLSToExecutionChange objects to node's pool and if passes validation node MUST broadcast it to network.
    */
-  submitPoolBlsToExecutionChange(
-    blsToExecutionChange: capella.SignedBLSToExecutionChange[]
-  ): Promise<ApiClientResponse<{[HttpStatusCode.OK]: void}, HttpStatusCode.BAD_REQUEST>>;
+  submitPoolBLSToExecutionChange: Endpoint<
```

Comment:
I think we should defer changing `Bls` to `BLS` from this PR and do it project wise later.

Reply:
This rename is required to make the routes testable against the spec which is critical for this PR to ensure correctness.

I don't feel strongly about renaming this, it's just required here to match spec `operationId`.

We could open an issue to do a follow-up renaming after this PR is merged.